### ISSUE TITLE
Project, Plan: keep one

### ISSUE THREAD

**Issue by shykes** [dagger/dagger#2462]
## Problem

Dagger defines two slightly different concepts: "project" and "plan". This made sense in 0.1 with `dagger up`, but in 0.2 with `dagger do`, it is less needed and adds unnecessary complexity to the developer experience.

## Solution

Merge the two concepts, and keep only word. I propose to keep the word "project" because it is more familiar to developers.

### How it works today

* A *project* is a shared context for developping and running Dagger actions. Often it maps to a developer's git repository, but that is not mandatory. Dagger uses the project to 1) load CUE files, and 2) store information about CUE dependencies.
* A *plan* is a directory within a project which can be loaded. Often the project directory is a valid plan (the project's "root plan"), but any sub-directory could also be a plan.

### What changed?

In 0.2, we deprecated `dagger up` and introduced `dagger do`. This made a big difference:

* With `dagger up`, users had no choice but to run an entire configuration. So to run different workflows (deploy to staging, run integration tests..) you had to maintain different plans.
* But now with `dagger do`, users only run one action at a time. So all workflows can co-exist in the same plan. Why bother maintaining different plans when users will need to select an action anyway?

In other words, with Dagger 0.2 a first-class concept of "plan" is unnecessary

### What about power users?

Power users can still use the same CUE layered loading feature. It will all work exactly the same, we just won't use the name "plan". Instead we'll say "Dagger supports all CUE features, including layered loading. If you load the project from one of its subdirectories, all files in the same CUE package will be layered together following the loading CUE standard. See [advanced CUE feature: layered loading] to learn more).

**Comment by shykes** [comment]
Assigning to myself because I think it's important and I don't want to lose track of it. cc @mircubed

**Comment by cdhunt** [comment]
In CI/CD products, `Pipeline`,  `Workflow`, and `Job` are defacto standard nouns. `Project` matches what .Net uses and `csproj` files work very similar to `dagger.#Plan`. I'm not sure there is a best choice, but I agree with picking one.

**Comment by mircubed** [comment]
@shykes based on comments above, do you have a final decision for this?

**Comment by mircubed** [comment]
Note to self- we need someone to work on this other than @shykes , so removing as assignee.

**Comment by dolanor** [comment]
I guess based on https://github.com/dagger/dagger/pull/2567#issuecomment-1211509736, we can close this one as well.
cc: @mircubed
